### PR TITLE
add function reshape to NDArray in scalapkg

### PR DESCRIPTION
**Pull request by Ldpe2G**

### CODE REVIEW

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/LibInfo.scala [hunk 1]
```diff
@@ -49,6 +49,10 @@ class LibInfo {
                              start: MXUint,
                              end: MXUint,
                              sliceHandle: NDArrayHandleRef): Int
+  @native def mxNDArrayReshape(handle: NDArrayHandle,
+                             nDim: Int,
+                             dims: Array[Int],
+                             reshapeHandle: NDArrayHandleRef): Int
```
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LibInfo.scala]: Please align the indent

File: scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc [hunk 2]
```diff
@@ -216,6 +216,16 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxNDArraySlice
   return ret;
 }
 
+JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxNDArrayReshape
+  (JNIEnv *env, jobject obj, jlong ndArrayPtr, jint ndim, jintArray dims, jobject reshapedHandle) {
+  NDArrayHandle out;
+  jint *pdims = env->GetIntArrayElements(dims, NULL);
```
Comment (javelinjs) [inline scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc]: remember to call ReleaseIntArrayElements when `pdims` is out of use.

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/NDArray.scala [hunk 3]
```diff
@@ -703,6 +703,19 @@ class NDArray private[mxnet](private[mxnet] val handle: NDArrayHandle,
   }
 
   /**
+   * Return a reshaped NDArray that shares memory with current one.
+   *
+   * @param dims New shape.
+   *
+   * @return a reshaped NDArray that shares memory with current one.
+   */
+  def reshape(dims: Array[Int]): NDArray = {
```
Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/NDArray.scala]: Please add a test case for this function.

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/LibInfo.scala (hunk 1 above)
Comment (Ldpe2G) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/LibInfo.scala]: fixed

File: scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc (hunk 2 above)
Comment (Ldpe2G) [inline scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc]: fixed

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/NDArray.scala (hunk 3 above)
Comment (Ldpe2G) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/NDArray.scala]: fixed

**Pull request merged**